bin/gvs

```diff
@@ -95,30 +95,53 @@ for REPO in "${REPOSITORIES[@]}"; do
 
   # Checkout
   if $DO_CHECKOUT; then
-    # Delete composer dependencies from WooCommerce
-    if [ -f woocommerce.php ]; then
-      $VERBOSE_OUTPUT && echo "Delete composer packages from $(pwd)" || true
-      rm -f composer.lock && rm -rf vendor
-    fi
-
     # Checkout target
     git checkout --force --quiet "$REPO_TARGET"
 
-    # Install composer dependencies for WooCommerce
-    if [ -f woocommerce.php ]; then
-      ORIGINAL_VERSION=$(composer --version | cut -d " " -f 3)
-      if [ "${ORIGINAL_VERSION::1}" == "2" ]; then
-        echo "Switching composer version to install WooCommerce dependencies"
-        composer_self_update 1
-      fi
-      $VERBOSE_OUTPUT && echo "Install composer packages to $(pwd)" || true
-      composer install --no-suggest --no-dev --quiet
-      NEW_VERSION=$(composer --version | cut -d " " -f 3)
-      if [ "${ORIGINAL_VERSION::1}" != "${NEW_VERSION::1}" ]; then
-        echo "Switching composer version to install WooCommerce dependencies"
-        composer_self_update rollback
+    # Directories to look for a composer.json file in
+    declare -a dirs=("$REPO_PATH" "$REPO_PATH/plugins/woocommerce")
+    for dir in "${dirs[@]}"; do
+      if [ -f "$dir/composer.json" ]; then
+        $VERBOSE_OUTPUT && echo "composer.json exists at $dir/composer.json"
+        cd "$dir"
+
+        # Clean up files and then install composer dependencies
+
+        if [ -f "$dir/composer.lock" ] && [ $(git ls-files composer.lock | wc -l) -eq "0" ]; then
+          $VERBOSE_OUTPUT && echo "$dir/composer.lock is not committed to git. Deleting."
+          rm -f "$dir/composer.lock"
+        fi
+
+        if [ -d "$dir/plugins" ] && [ $(git status --porcelain | grep plugins/ | wc -l) -ne "0" ]; then
+          rm -rf "$dir/plugins"
+          $VERBOSE_OUTPUT && echo "$dir/plugins from WC monorepo deleted."
+        fi
+        if [ -d "$dir/vendor" ]; then
+          $VERBOSE_OUTPUT && echo "$dir/vendor exists"
+          if [ $(git ls-files vendor | wc -l) -ne "0" ]; then
+            $VERBOSE_OUTPUT && echo "$dir/vendor is committed to git, so no composer install required. Skipping."
+            break
```

Comment:
Should not be this the first thing to check?

Also can the deletion logic can be combine to the rest (see comment on lines +110 to +113).

Reply:
I think in this instance even if `/vendor/` exists and is committed, there still may be cleanup required for an uncommitted `composer.lock` file.

Use case: switching from an older WooCommerce such as 4.9.0 (which [has vendor and composer.lock both committed](https://github.com/woocommerce/woocommerce/tree/4.9.0)), to a new version such as 5.9.0, which has neither committed. In that case no composer install is required, but composer.lock should also be deleted and not left on the filesystem.

So my vote is to leave this code section as-is.